Dart/src/com/jetbrains/lang/dart/ide/editor/DartEnterInStringHandler.java

```diff
@@ -31,37 +31,30 @@ public Result preprocessEnter(@NotNull final PsiFile file,
     Document document = editor.getDocument();
     ASTNode token = psiAtOffset.getNode();
     IElementType type = token.getElementType();
+    if (type == DartTokenTypes.RAW_TRIPLE_QUOTED_STRING) {
+      return Result.DefaultSkipIndent; // Multiline string gets no indent
+    }
     if (type == DartTokenTypes.RAW_SINGLE_QUOTED_STRING) {
       char quote = token.getText().charAt(1);
       breakString("r" + quote, String.valueOf(quote), caretOffsetRef, caretAdvanceRef, document);
       return Result.Default;
     }
-    if (type == DartTokenTypes.REGULAR_STRING_PART) {
-      token = token.getTreeNext();
-      if (token == null) return Result.Continue; // Can't happen with current grammar.
+    while ((token = token.getTreeParent()) != null) {
```

Comment:
Do you think we can optimize it by checking not more than 2 parents up the hierarchy?